Applications/working_with_events/app.py

```diff
@@ -1,62 +1,55 @@
 import time
-from threading import Event
+import logging as log
 
-import depthai as dai
+# cv2 and av bug workaround
+import cv2
+import numpy as np
 
-import logging as log
+cv2.imshow("bugfix", np.zeros((10, 10, 3), dtype=np.uint8))
+cv2.destroyWindow("bugfix")
+
+import depthai as dai
 import robothub as rh
 
 from pipeline import create_pipeline
+from handlers import EventHandler
 
 
 class Application(rh.BaseDepthAIApplication):
 
     def __init__(self):
         super().__init__()
         self.detection_view = rh.DepthaiLiveView(name="detection_view", unique_key="rgb", width=1920, height=1080)
-        self.last_upload = None
-        self.take_picture_signal = Event()
-        rh.COMMUNICATOR.on_frontend(notification=self.on_fe_notification)
+        self.event_handler = EventHandler()
 
     def setup_pipeline(self) -> dai.Pipeline:
         pipeline = dai.Pipeline()
-        create_pipeline(pipeline=pipeline, config=self.config)
+        create_pipeline(pipeline=pipeline, config=rh.CONFIGURATION)
         return pipeline
 
-    def on_fe_notification(self, session_id, unique_key, payload):
-        if unique_key == 'take_picture':
-            print("Received take picture notification from FE")
-            self.take_picture_signal.set()
-
     def manage_device(self, device: dai.Device):
         log.info(f"DepthAi version: {dai.__version__}")
         log.info(f"Oak started. getting queues...")
         rgb_h264 = device.getOutputQueue(name="rgb_h264", maxSize=5, blocking=False)
         rgb_mjpeg = device.getOutputQueue(name="rgb_mjpeg", maxSize=5, blocking=False)
         detection_nn = device.getOutputQueue(name="detection_nn", maxSize=5, blocking=False)
 
-        while self.running:
+        while rh.app_is_running:
             rgb_h264_frame: dai.ImgFrame = rgb_h264.get()
             rgb_mjpeg_frame: dai.ImgFrame = rgb_mjpeg.get()
             detections: dai.ImgDetections = detection_nn.get()
-            for detection in detections.detections:
-                current_time = time.time()
-                if self.last_upload and current_time - self.last_upload < self.config["event_upload_interval"]:
-                    continue
-                if detection.label == 0:
-                    self.last_upload = current_time
-                    rh.events.send_image_event(image=rgb_mjpeg_frame.getFrame(), title="Person detected",
-                                               device_id=device.getMxId())
-            if self.take_picture_signal.is_set():
-                self.take_picture_signal.clear()
-                log.info(f"Sending Event...")
-                rh.events.send_image_event(image=rgb_mjpeg_frame.getFrame(), title="Front-end event image",
-                                           device_id=device.getMxId())
+
+            self.event_handler.send_spaced_img_event_on_detection(detections=detections,
```

Comment:
That's true. The detection function sends the events on a certain interval, so I was trying to incorporate that into the name. Do you have a better suggestion, that would do that or is `send_event_on_detection` fine?

Reply:
Yes. So the question is how detailed explanation should the function name provide. Should it include “img”, should it contain the information about the frequency?
I leave it up to you, i am just asking for consistency. Ie if you use img_detection for one then you have to use it for the other as well since both send the same detection type